### PR TITLE
Add wiki paragraph describing supported Rez config file types

### PR DESCRIPTION
**Pull request by brycegbrazen**
Built the docs with Python 3.11.6.

### CODE REVIEW

**Comment by brycegbrazen** [comment]
Note: I'd like to specify the package versions in the `docs/requirements.txt` file, but I'm unsure exactly which versions of these packages were used in the initial build of the documentation. Let me know if anyone can provide this.

Looking to improve the file to say something like:
```
sphinx~=1.0
furo~=1.0
sphinx-autobuild~=1.0
```

**Comment by brycegbrazen** [comment]
Also, I'm unsure what the `docs/NOTES.md` file is supposed to contain or if this was a leftover file from someone. Perhaps it would be better to replace this file with a `docs/BUILD.md` that explains the process for building the docs? I would imagine the `BUILD.md` file would detail the supported Python versions for the build.

File: docs/source/configuring_rez.rst [hunk 1]
```diff
@@ -28,6 +28,14 @@ variable :envvar:`REZ_CONFIG_FILE` is then set to for all your users.
...
 .. _configuring-rez-settings-merge-rules:
 
+Supported Configuration File Types
+====================
+
+Rez supports both YAML configuration files and Python configuration files.
+
+You may prefer a Python based configuration file if you need to vary your configuration settings based on your
+current platform.
+
```
Comment (JeanChristopheMorinPerso) [inline docs/source/configuring_rez.rst:38]:
```suggestion 
Supported Configuration File Formats
====================================

Rez supports both YAML configuration files and Python configuration files.

You may prefer a Python based configuration file if you need to vary your configuration settings based on your
current platform.

.. _configuring-rez-settings-merge-rules:

```

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

**Comment by brycegbrazen** [comment]
With Sphinx 7.2.6, and Python 3.9, I get the following error:

```
Exception occurred:
  File "D:\Code_forks\rez\docs\rez_sphinxext.py", line 339, in <module>
    def setup(app: sphinx.application.Sphinx) -> dict[str, bool | str]:
TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

Looks like Python 3.10+3.11 works fine though.

File: docs/source/configuring_rez.rst [hunk 2]
```diff
@@ -26,8 +26,6 @@ variable :envvar:`REZ_CONFIG_FILE` is then set to for all your users.
    You do not need to provide a copy of all settings in this file. Just provide those
    that are changed from the defaults.
 
-.. _configuring-rez-settings-merge-rules:
-
 Supported Configuration File Types
 ====================
```
Comment (JeanChristopheMorinPerso) [inline docs/source/configuring_rez.rst:30]:
```suggestion
Supported Configuration File Formats
====================================
```

**Comment by JeanChristopheMorinPerso** [comment]
>Note: I'd like to specify the package versions in the docs/requirements.txt file, but I'm unsure exactly which versions of these packages were used in the initial build of the documentation.

I kept them open for now because constraining them will require us to enable dependabot to make sure we keep these dependencies up to date. Eventually we'll do it.

>Also, I'm unsure what the docs/NOTES.md file is supposed to contain or if this was a leftover file from someone. Perhaps it would be better to replace this file with a docs/BUILD.md that explains the process for building the docs? I would imagine the BUILD.md file would detail the supported Python versions for the build.

`NOTES.md` was a way for me to remember what characters I used for the headers. Feel free to rename it `README.md` and add instruction on how to build the docs. I'd like to keep the example headers though.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Thanks @brycegbrazen !

**Pull request merged**